tests/integration/operator/switch_cr_test.go

```diff
@@ -115,61 +117,50 @@ func TestController(t *testing.T) {
 			installWithCRFile(t, ctx, cs, s, istioCtl, "default", "v2")
 
 			verifyInstallation(t, ctx, istioCtl, "default", "v2", cs)
-
-			t.Cleanup(func() {
-				scopes.Framework.Infof("cleaning up resources")
-				if err := cs.DeleteYAMLFiles(IstioNamespace, iopCRFile); err != nil {
-					t.Errorf("failed to delete test IstioOperator CR: %v", err)
-				}
-				if err := cs.AppsV1().Deployments(IstioNamespace).DeleteCollection(context.TODO(),
-					kube2.DeleteOptionsForeground(), kubeApiMeta.ListOptions{LabelSelector: "app=istiod"}); err != nil {
-					t.Errorf("failed to remove istiod deployments: %v", err)
-				}
-			})
-		})
-}
-
-func TestOperatorRemove(t *testing.T) {
-	framework.
-		NewTest(t).
-		Run(func(ctx framework.TestContext) {
-			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
-			cs := ctx.Clusters().Default()
-			s, err := image.SettingsFromCommandLine()
-			if err != nil {
-				t.Fatal(err)
-			}
-			initCmd := []string{
-				"operator", "init",
-				"--hub=" + s.Hub,
-				"--tag=" + s.Tag,
-				"--manifests=" + ManifestPath,
-			}
-			istioCtl.InvokeOrFail(t, initCmd)
-
+			// test operator remove command
+			scopes.Framework.Infof("checking operator remove command")
+			cleanupInClusterCRs(t, cs)
 			removeCmd := []string{
 				"operator", "remove",
 			}
-			// install second operator deployment with different revision
 			istioCtl.InvokeOrFail(t, removeCmd)
-			retry.UntilSuccessOrFail(t, func() error {
-				if svc, _ := cs.CoreV1().Services(OperatorNamespace).Get(context.TODO(), "istio-operator", kubeApiMeta.GetOptions{}); svc.Name != "" {
-					return fmt.Errorf("got operator service: %s from cluster, expected to be removed", svc.Name)
-				}
 
-				if dp, _ := cs.AppsV1().Deployments(OperatorNamespace).Get(context.TODO(), "istio-operator", kubeApiMeta.GetOptions{}); dp.Name != "" {
-					return fmt.Errorf("got operator deploymentL %s from cluster, expected to be removed", dp.Name)
+			retry.UntilSuccessOrFail(t, func() error {
+				for _, n := range []string{"istio-operator", "istio-operator-v2"} {
+					if svc, _ := cs.CoreV1().Services(OperatorNamespace).Get(context.TODO(), n, kubeApiMeta.GetOptions{}); svc.Name != "" {
+						return fmt.Errorf("got operator service: %s from cluster, expected to be removed", svc.Name)
+					}
+					if dp, _ := cs.AppsV1().Deployments(OperatorNamespace).Get(context.TODO(), n, kubeApiMeta.GetOptions{}); dp.Name != "" {
+						return fmt.Errorf("got operator deployment %s from cluster, expected to be removed", dp.Name)
+					}
 				}
 				return nil
 			}, retry.Timeout(retryTimeOut), retry.Delay(retryDelay))
 
-			// cleanup created resources
 			t.Cleanup(func() {
 				scopes.Framework.Infof("cleaning up resources")
+				unInstallCmd := []string{
+					"x", "uninstall", "--purge", "--skip-confirmation",
```

Comment:
this doesn't match https://istio.io/latest/docs/setup/install/operator/#uninstall, should the doc be updated?

Reply:
this corresponds to this section instead: https://istio.io/latest/docs/setup/install/istioctl/#uninstall-istio